docs/developer/architecture/snmp.md

```diff
@@ -103,6 +106,199 @@ snmp_listener:
 
 ### Cluster Agent Support
 
-For Kubernetes environments, the [Cluster Agent](https://docs.datadoghq.com/agent/cluster_agent/) can be configured to use the Agent auto-discovery logic as a source of [Cluster checks](https://docs.datadoghq.com/agent/cluster_agent/clusterchecks/).
+For Kubernetes environments, the [Cluster Agent](https://docs.datadoghq.com/agent/cluster_agent/) can be configured to use the SNMP Agent auto-discovery (uses snmp listener) logic as a source of [Cluster checks](https://docs.datadoghq.com/agent/cluster_agent/clusterchecks/).
+
+![](../../assets/images/snmp-architecture-cluster-agent.png)
+
+The Datadog Cluster Agent (DCA) uses the `snmp_listener` config (Agent auto-discovery) to listen for IP ranges, then schedule snmp check instances to be run by one or more normal Datadog Agents.
+
+Agent auto-discovery combined with Cluster Agent is very scalable, it can be used to monitor a large number of snmp devices.
+
+#### Example Cluster Agent setup with SNMP Agent auto-discovery using Datadog helm-chart
+
+First you need to [add Datadog Helm repository](https://github.com/DataDog/helm-charts).
```

Comment:
```suggestion
First you need to [clone the Datadog Helm repository](https://github.com/DataDog/helm-charts).
```

Do you mean clone?

Reply:
I think `add` is more appropriate since the command to add the repository is:

```
helm repo add datadog https://helm.datadoghq.com
```

Reply:
Ahh cool, can we add that command as well?